src/execution_plan/ops/op_merge.c

```diff
@@ -247,6 +247,11 @@ static Record MergeConsume(OpBase *opBase) {
 				lhs_record = NULL;
 			}
 			must_create_records = true;
+			/* Since op_merge is is closer to the results op then op_merge_create, it is initially may be marked as the last writer
+			 * Now that it knows that op_merge_create will write after the merge_op write it should try to replace itself as the
+			 * last writer. */
+
+			QueryCtx_UpdateLastWriter((OpBase *)op, op->create_stream);
```

Comment:
Instead of having this here, I'd put the following before the call to `_UpdateProperties` at 266:
`if(must_create_records) QueryCtx_UpdateLastWriter(opBase, op->create_stream);`

Then it will only execute once.

Reply:
done